Startup/startup_ch32l103.S

```diff
@@ -0,0 +1,269 @@
+    .weak   USART3_IRQHandler          /* USART3 */
+    .weak   EXTI15_10_IRQHandler       /* EXTI Line 15..10 */
+    .weak   RTCAlarm_IRQHandler        /* RTC Alarm through EXTI Line */
+    .weak   LPTIMWakeUp_IRQHandler     /* LPTIM Wake up */
+    .weak   USBFS_IRQHandler           /* USBFS */
+    .weak   USBFSWakeUp_IRQHandler     /* USBFS Wake Up */
+    .weak   USART4_IRQHandler          /* USART4 */
+    .weak   DMA1_Channel8_IRQHandler   /* DMA1 Channel8 */
+    .weak   LPTIM_IRQHandler           /* LPTIM */
+    .weak   OPA_IRQHandler             /* OPA */
+    .weak   USBPD_IRQHandler           /* USBPD */
+    .weak   USBPDWakeUp_IRQHandler     /* USBPD Wake Up */
+    .weak   CMPWakeUp_IRQHandler       /* CMP Wake Up */
+
+NMI_Handler:
+HardFault_Handler:
+Ecall_M_Mode_Handler:
+Ecall_U_Mode_Handler:
+Break_Point_Handler:
+SysTick_Handler:
+SW_Handler:
+WWDG_IRQHandler:
+PVD_IRQHandler:
+TAMPER_IRQHandler:
+RTC_IRQHandler:
+FLASH_IRQHandler:
+RCC_IRQHandler:
+EXTI0_IRQHandler:
+EXTI1_IRQHandler:
+EXTI2_IRQHandler:
+EXTI3_IRQHandler:
+EXTI4_IRQHandler:
+DMA1_Channel1_IRQHandler:
+DMA1_Channel2_IRQHandler:
+DMA1_Channel3_IRQHandler:
+DMA1_Channel4_IRQHandler:
+DMA1_Channel5_IRQHandler:
+DMA1_Channel6_IRQHandler:
+DMA1_Channel7_IRQHandler:
+ADC_IRQHandler:
+USB_HP_CAN1_TX_IRQHandler:
+USB_LP_CAN1_RX0_IRQHandler:
+CAN1_RX1_IRQHandler:
+CAN1_SCE_IRQHandler:
+EXTI9_5_IRQHandler:
+TIM1_BRK_IRQHandler:
+TIM1_UP_IRQHandler:
+TIM1_TRG_COM_IRQHandler:
+TIM1_CC_IRQHandler:
+TIM2_IRQHandler:
+TIM3_IRQHandler:
+TIM4_IRQHandler:
+I2C1_EV_IRQHandler:
+I2C1_ER_IRQHandler:
+I2C2_EV_IRQHandler:
+I2C2_ER_IRQHandler:
+SPI1_IRQHandler:
+SPI2_IRQHandler:
+USART1_IRQHandler:
+USART2_IRQHandler:
+USART3_IRQHandler:
+EXTI15_10_IRQHandler:
+RTCAlarm_IRQHandler:
+LPTIMWakeUp_IRQHandler:
+USBFS_IRQHandler:
+USBFSWakeUp_IRQHandler:
+USART4_IRQHandler:
+DMA1_Channel8_IRQHandler:
+LPTIM_IRQHandler:
+OPA_IRQHandler:
+USBPD_IRQHandler:
+USBPDWakeUp_IRQHandler:
+CMPWakeUp_IRQHandler:
+1:
+	j 1b
+
+	.section	.text.handle_reset,"ax",@progbits
+	.weak	handle_reset
+	.align	1
+handle_reset:
+.option push 
+.option	norelax 
+	la gp, __global_pointer$
+.option	pop 
+1:
+	la sp, _eusrstack 
+2:
+/* Load data section from flash to RAM */
+	la a0, _data_lma
+	la a1, _data_vma
+	la a2, _edata
+	bgeu a1, a2, 2f
+1:
+	lw t0, (a0)
+	sw t0, (a1)
+	addi a0, a0, 4
+	addi a1, a1, 4
+	bltu a1, a2, 1b
+2:
+/* Clear bss section */
+	la a0, _sbss
+	la a1, _ebss
+	bgeu a0, a1, 2f
+1:
+	sw zero, (a0)
+	addi a0, a0, 4
+	bltu a0, a1, 1b
+2:
+/* Configure pipelining and instruction prediction */
+    li t0, 0x1f
+    csrw 0xbc0, t0
+/* Enable interrupt nesting and hardware stack */
+	li t0, 0x3
+	csrw 0x804, t0
+/* Enable global interrupt and configure privileged mode */
+   	li t0, 0x88           
+   	csrw mstatus, t0
+/* Configure the interrupt vector table recognition mode and entry address mode */
+ 	la t0, _vector_base
+    ori t0, t0, 3           
+	csrw mtvec, t0
```

Comment:
Technically this doesn't contain the usual modifications we do to the startup files. Thus RTOS frameworks like FreeRTOS, LiteOS, RTThread and TencentOS won't work, as well as C++ object initilization (`__libc_init_array`). We'll just add that later...

Reply:
Tracked https://github.com/Community-PIO-CH32V/framework-wch-noneos-sdk/issues/7

Reply:
Thank you for review.
I apologize for any improprieties in the code.

Reply:
You couldn't have known that we do such hacks. To be honest, it might be cleaner to use the dedicated linker script for each package. The only reason this is done here is so that features can be turned on/off in a unified way and the flash and RAM size can be automatically substituted -- they're always hardcoded in the WCH given `Link.ld` files.